kubernetes/utils/deserialize.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import pydoc
+import re
+import sys
+from os import path
+
+import yaml
+
+from kubernetes import client
+
+PYDOC_RETURN_LABEL = ":return:"
+
+
+class RespMock(object):
+    """
+    A dummy class to mock RESTResponse object
+    """
+
+    def __init__(self, *args):
+        self.data = None
+
+
+def load_from_json(json_file, klass=None, verbose=False):
+    """
+    Load objects from json file. Pass True for verbose to
+    print additional information.
+    Input:
+    json_file: string. Contains the path to json file.
+    klass: class literal for deserialized object,
+    or string of class name.
+    verbose: If True, print information about the object lookup info.
+        Default is False.
+
+    Raises:
+        FailToLoadError which holds list of load failures..
+    """
+    with open(path.abspath(json_file)) as file:
+        data = file.read()
+        # python 2 has problems decoding unicode from json.load
+        # use yaml.load instead
+        if sys.version_info[0] < 3:
+            data = yaml.safe_load(data)
+        else:
+            data = json.loads(data)
+        obj = load_from_dict(data, klass=klass, verbose=verbose)
+
+        return obj
+
+
+def load_from_yaml(yaml_file, klass=None, verbose=False):
+    """
+    Perform an action from a yaml file. Pass True for verbose to
+    print additional information.
+    Input:
+    yaml_file: string. Contains the path to yaml file.
+    klass: class literal for deserialized object,
+    or string of class name.
+    verbose: If True, print information about the object lookup info.
+        Default is False.
+
+    Raises:
+        FailToLoadError which holds list of load failures.
+    """
+    with open(path.abspath(yaml_file)) as f:
+        yml_document_all = yaml.safe_load_all(f)
+        failures = []
+        objs = []
+        for yml_document in yml_document_all:
+            try:
+                obj = load_from_dict(
+                    yml_document, klass=klass, verbose=verbose)
+                # Prevent returning list of lists when doing multi
+                # document yaml
+                if isinstance(obj, list):
+                    objs.extend(obj)
+                else:
+                    objs.append(obj)
```

Comment:
Could you justify the use case for deserializing a yaml list into a list of objects?

Reply:
I don't think I can justify it if we can handle list type as you suggested above. I will explore that option first and this shall be removed if that is implemented.